backend/app/server/database.py

```diff
@@ -14,4 +14,8 @@ async def init():
     client = motor.motor_asyncio.AsyncIOMotorClient(MONGO_URL)
     database = client.get_database(DB_NAME)
 
-    await init_beanie(database=database, document_models=[Shop])
+    # Create a 2dsphere index on the 'location' field of the 'Shop' collection
+    shop_collection = database.get_collection('Shop')
+    await shop_collection.create_index([("location", "2dsphere")])
```

Comment:
index dodawaj w klasie Shop (możesz całą kolejke usunąć w razie czego), tak jak w tym tutorialu https://dev.to/romanright/mongodb-indexes-with-beanie-43e8